### PR TITLE
Adding __citation__ for the main astropy reference 

### LINKED ISSUE #6126: reads CITATION file from cwd
When you import the `astropy` module, it will read the `CITATION` file from the current working directory (if such file exists). I believe this was meant to read astropy's own citation file. But current working directory can't be assumed to contain the astropy source. Local users could exploit this bug for denial of service if a program that uses Astropy is run in an ustrusted directory (such as `/tmp`).

### PR DESCRIPTION
**Pull request by bsipocz**
@adrn - I think having this would make sense as the users may discover it sooner, but with a quick search showed there are order of mag more packages with ``__bibtex__``. 

There is then also an option that we may want ``__citation__`` and ``__bibtex__`` have different meaning, but I don't see that would do much good.

fixes #6126

### CODE REVIEW

**Comment by saimn** [comment]
`__citation__` seems better indeed. 
But also a reminder for #6126: this works only when importing astropy from the root of a source directory! I started something for this some time ago but forgot about it, one issue is that the CITATION file is not installed currently.

File: astropy/__init__.py [hunk 1]
```diff
@@ -275,6 +275,7 @@ def _get_bibtex():
 
 __bibtex__ = _get_bibtex()
 
+__citation__ = _get_bibtex()
```
[the remark this answers is not in the document]
Comment (bsipocz) [reply astropy/__init__.py]: sure, that was my original :)

**Comment by bsipocz** [comment]
@saimn - thanks for the reminder. It definitely needs to go into the manifest, I'll double check that this keep working once being packaged up.

**Comment by saimn** [comment]
@bsipocz - The file must be moved in the `astropy/` directory (package), otherwise it will not be installed. But this makes it hidden when browsing files from the git repository, which is not ideal either.

**Comment by bsipocz** [comment]
So, this is a little bit ugly, but I'm confident that a workable workaround that both looks OK on github and provides a `__citation__` and `__bibtex__` method with the released/installed version (actually I've checked this, the only missing thing is that I'm not sure how the symlink work on Windows).

File: astropy/__init__.py [hunk 2]
```diff
@@ -263,18 +263,20 @@ def _rebuild_extensions():
...
-            refs = re.findall(r'\{[^()]*\}', citation.read())
-            if len(refs) == 0: return ''
-            bibtexreference = "@ARTICLE{0}".format(refs[0])
-        return bibtexreference
-    else:
-        return ''
+
+    citation_file = os.path.join(os.path.dirname(__file__), 'CITATION')
+
+    print(citation_file)
```
Comment (astrofrog) [inline astropy/__init__.py]: Leftover from debugging?

**Comment by pllim** [comment]
In general, symlinks don't work on Windows.

> But this makes it hidden when browsing files

I agree. Is it too much to retain a top-level `CITATION` even if the content is only "see `astropy/CITATION`" ?

File: astropy/__init__.py (hunk 2 above)
Comment (bsipocz) [reply astropy/__init__.py]: oh, indeed.

**Comment by bsipocz** [comment]
> I agree. Is it too much to retain a top-level CITATION even if the content is only "see astropy/CITATION" ?

The `CITATION` file is a weird one, people are still talking about what format is should be what metadata it should contain, etc. 
I think your suggestion makes sense, much more sense than repeating the same file as then I would fear for them to fall out of sync. But then I don't know whether any tools are looking for these `CITATION` files yet and whether they would fall over on this.

**Comment by bsipocz** [comment]
@pllim - that is weird indeed. Maybe I shouldn't have had the move of the file and the symlink addition in  the same commit, it usually picks up nicely is a file was moved rather than deleted and created.

**Comment by bsipocz** [comment]
I merged this as the zeroth order approach. I kept the original workaround of having a symlink, but please feel free to update the `CITATION` file to rather point to `astropy/CITATION` (It's not clear to me whether any indexing relies on the content of `CITATION` or not yet, and even if is what kind of metadata is required, etc.).

**Pull request merged**